nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -605,35 +659,143 @@ sub multiLineList {
+  # kernel. Also see `sound.mediaKeys.enable` in configuration.nix(5).
+  #
   # sound.enable = true;
+
+  # Enable sound output via PulseAudio. This option will enable
+  # PulseAudio daemon and configure `asound.conf` to route the
+  # `default` ALSA pcm and ctl to PulseAudio daemon.
+  #
   # hardware.pulseaudio.enable = true;
+  #
+  # Note that you don't need to enable PulseAudio daemon to get either
+  # of sound playback or software sound mixing. It will out of the box
+  # over ALSA. If you do want to run PulseAudio daemon, however, you
+  # may safely disable `sound.enable`.
+
+  # Also note that you can emulate PulseAudio API over ALSA for apps
+  # that refuse to work over ALSA themselves by running them under
+  # apulse. This way you can run one daemon less which benefits your
+  # security (some people claim that PulseAudio codebase lacks in
+  # quality to be connected to untrusted data sources like the browser
+  # by default).
+  #
+  # environment.systemPackages = [ pkgs.apulse ];
 
   # Enable the X11 windowing system.
+  #
   # services.xserver.enable = true;
   # services.xserver.layout = "us";
   # services.xserver.xkbOptions = "eurosign:e";
-
+  #
   # Enable touchpad support.
   # services.xserver.libinput.enable = true;
 
+  # Enable graphical acceleration in X11. See configuration.nix(5) man
+  # page for possible options.
+  #
+  # services.xserver.videoDrivers = [ "<name-of-the-driver>" ];
+
+  # Enable support for 32-bit OpenGL libraries on x86_64 which is
+  # useful if you want to run 32-bit OpenGL applications
+  #
+  # hardware.opengl.driSupport32Bit = true;
+  #
+  # E.g., games under Wine
+  #
+  # environment.systemPackages = with pkgs; [ wine winetricks ];
+
+  # Some of the X11 drivers above might not work with the current Linux
+  # kernel, so you might end up overriding that too.
+  #
+  # boot.kernelPackages = pkgs.linuxPackages_4_9;
+
+  # Define a user account. Don't forget to set a password with ‘passwd’.
+  #
+  # users.users.me = {
+  #   isNormalUser = true;
+  #   uid = 1000;
+  #   extraGroups = [ "input" "audio" "video" "cdrom" ]
+  #              ++ [ "wheel" ]; # for sudo
+  #   openssh.authorizedKeys.keyFiles = [ /etc/nixos/id_rsa.pub ];
+  # };
+
   # Enable the KDE Desktop Environment.
+  #
   # services.xserver.displayManager.sddm.enable = true;
   # services.xserver.desktopManager.plasma5.enable = true;
 
-  # Define a user account. Don't forget to set a password with ‘passwd’.
-  # users.users.guest = {
+  # Or maybe you want to enable Xfce Desktop Environment and Slim
+  # display manager?
+  #
+  # services.xserver.displayManager.slim.enable = true;
+  # services.xserver.desktopManager.xfce.enable = true;
+
+  # Or maybe you just want to automatically login into your .xsession
+  # script? Then don't enable any desktopManagers and uncomment the
+  # following.
+  #
+  # services.xserver.displayManager.auto = {
+  #   enable = true;
+  #   user = "me";
+  # };
```

Comment:
Maybe this whole section could be marked as "for desktop use only"? Move the user stuff below it though.